File: ElectronicNoteBookTask/ElectronicNoteBookTask/AddNoteCommand.cs
namespace ElectronicNoteBookTask
{
    /// <summary>
    /// Class of specific command.
    /// A reference to the NoteBookProvider is stored 
    /// which the command will redirect the main job to NoteBookProvider.
    /// </summary>
    public class AddNoteCommand: ICommand
    {
        private NoteBookProvider noteBookProvider;
        private NoteBook noteBook;
        private Note note;

        /// <summary>
        /// Constructor assigns the value to the fields.
        /// </summary>
        /// <param name="provider">Field provider gets the value of parameter provider</param>
        /// <param name="noteBook">Field noteBook gets the value of parameter notebook</param>
        /// <param name="note">Field notegets the value of parameter note</param>
        public AddNoteCommand(NoteBookProvider provider, NoteBook noteBook, Note note)
        {
            noteBookProvider = provider;
            this.noteBook = noteBook;
            this.note = note;
        }

        /// <summary>
        /// Method evoke for executing a specific method which add note to notebook.
        /// </summary>
        public void Execute()
        {
           noteBook = noteBookProvider.AddNote(noteBook, note);
        }
    }
}

File: NoteBook/src/main/java/AddNoteCommand.java
import java.util.Date;

/**
 * This class realize add Note to notebook.
 */

public class AddNoteCommand implements Command{
  NoteBook noteBook;
  String value;
  public AddNoteCommand(NoteBook noteBook, String value) {
    this.value = value;
    this.noteBook = noteBook;
  }
  public void setValue(String value) {
    this.value = value;
  }

  /**
   * This method override method execute from Command interface for addNote method.
   */
  public void execute() {
    Date date = new Date();
    Note note = new Note(value, date);
    noteBook.getNotes().add(note);
  }
}